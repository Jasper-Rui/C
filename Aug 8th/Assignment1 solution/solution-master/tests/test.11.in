TRX 11235 Alice_Q_Public Bob_X_Private 150000 20
TRX 12358 Carol_Y_Incredible Dave_Z_Particle 350000 40
TRX 23581 Eve_N_Star Fred_F_Filibuster 1350000 10
TRX 35813 Gina_T_Positron Hugh_H_Haliboo 11350000 100
BLK 1 0 0x00000000 2
11235 Alice_Q_Public Bob_X_Private 150000 20
23581 Eve_N_Star Fred_F_Filibuster 1350000 10
0x000000ec 0x00c64a6a
MINE
END

