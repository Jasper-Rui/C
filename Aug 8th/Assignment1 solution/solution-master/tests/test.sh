#!/bin/sh

echo ======================================================
echo ====================== TEST $1 ========================
echo ======================================================
if timeout 10 ./miner/miner < tests/test.$1.in > tests/test.$1.out; then 
  if diff tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    echo =====================
    echo Actual Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    exit 1
  fi
elif [ $? == 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
