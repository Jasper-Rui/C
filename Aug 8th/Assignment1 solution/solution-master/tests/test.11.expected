Adding transaction: 11235 Alice_Q_Public Bob_X_Private 150000 20
Adding transaction: 12358 Carol_Y_Incredible Dave_Z_Particle 350000 40
Adding transaction: 23581 Eve_N_Star Fred_F_Filibuster 1350000 10
Adding transaction: 35813 Gina_T_Positron Hugh_H_Haliboo 11350000 100
Removing transaction: 11235 Alice_Q_Public Bob_X_Private 150000 20
Removing transaction: 23581 Eve_N_Star Fred_F_Filibuster 1350000 10
Block mined: 2 1 0x00c64a6a 2
12358 Carol_Y_Incredible Dave_Z_Particle 350000 40
35813 Gina_T_Positron Hugh_H_Haliboo 11350000 100
0x00000078 0x009b107c
